engine_test.go

```diff
@@ -0,0 +1,38 @@
+package fuego
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithErrorHandler(t *testing.T) {
+	t.Run("default engine", func(t *testing.T) {
+		e := NewEngine()
+		err := NotFoundError{
+			Err: errors.New("Not Found :c"),
+		}
+		errResponse := e.ErrorHandler(err)
+		require.ErrorAs(t, errResponse, &HTTPError{})
+	})
+	t.Run("custom handler", func(t *testing.T) {
+		e := NewEngine(
+			WithErrorHandler(func(err error) error {
+				return errors.New("")
+			}),
+		)
+		err := NotFoundError{
+			Err: errors.New("Not Found :c"),
+		}
+		errResponse := e.ErrorHandler(err)
+		require.NotErrorAs(t, errResponse, &HTTPError{})
+	})
```

Comment:
Please add a test about a custom wrapper that wrap the initial one

```suggestion
	t.Run("custom handler", func(t *testing.T) {
		e := NewEngine(
			WithErrorHandler(func(err error) error {
				return fmt.Errorf("%w foobar", err)
			}),
		)
		err := NotFoundError{
			Err: errors.New("Not Found :c"),
		}
		errResponse := e.ErrorHandler(err)
		require.ErrorAs(t, errResponse, &HTTPError{})
		require.ErrorIs(t, errResponse, err)
		require.ErrorContains(t, errResponse, "foobar")
	})
```

Reply:
ermmm okay sure

Reply:
You suggestion doesn't pass.

Reply:
That the custom error handler is receiving the error and is able to manipulate it.

Reply:
Okay sure. I'll update the suggestion to work. fmt.ErrorF actually creates `wrapError` which is why the above doesn't work

Reply:
Updated